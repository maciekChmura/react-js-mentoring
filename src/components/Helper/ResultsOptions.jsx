import React, { Component } from 'react';
import styled from 'styled-components';
import SortingButton from './SortingButton';

const ResultsOptionsWrapper = styled.div`
  grid-area: 2 / 2 / 3 / 3;
  display: flex;
  justify-content: space-between;
`;

const InfoWrapper = styled.div``;

const ButtonsWrapper = styled.div`
  display: flex;
`;

const Display = styled.p`
  margin: 5px;
`;

class ResultsOptions extends Component {
  render() {
    const { dataSize } = this.props;
    return (
      <ResultsOptionsWrapper>
        <InfoWrapper>
          <Display>{dataSize} movies found</Display>
        </InfoWrapper>
        <ButtonsWrapper>
          <Display>Sort by:</Display>
          <SortingButton
            text="release date"
            handleClick={this.props.handleClick}
          />
          <SortingButton text="rating" handleClick={this.props.handleClick} />
        </ButtonsWrapper>
      </ResultsOptionsWrapper>
    );
  }
}

export default ResultsOptions;
