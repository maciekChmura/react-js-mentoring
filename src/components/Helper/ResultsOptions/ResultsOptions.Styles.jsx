import styled from 'styled-components';

export const ResultsOptionsWrapper = styled.div`
  grid-area: 2 / 2 / 3 / 3;
  display: flex;
  justify-content: space-between;
  background-color: #eee;
`;

export const ButtonsWrapper = styled.div`
  display: flex;
`;

export const Display = styled.p`
  font-size: 18px;
  margin: 5px;
  padding: 5px;
`;
