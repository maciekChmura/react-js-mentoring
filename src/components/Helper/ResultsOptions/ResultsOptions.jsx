import React from 'react';
import {
  ResultsOptionsWrapper,
  ButtonsWrapper,
  Display
} from './ResultsOptions.Styles';
import OptionButton from '../OptionButton/OptionButton';

const ResultsOptions = ({ dataSize, changeSorting, sortingType }) => (
  <ResultsOptionsWrapper>
    <Display>{dataSize} movies found</Display>
    <ButtonsWrapper>
      <Display>Sort by:</Display>
      {['release date', 'rating'].map(title => (
        <OptionButton
          text={title}
          changeOption={changeSorting}
          option={sortingType}
          key={title}
        />
      ))}
    </ButtonsWrapper>
  </ResultsOptionsWrapper>
);

export default ResultsOptions;
