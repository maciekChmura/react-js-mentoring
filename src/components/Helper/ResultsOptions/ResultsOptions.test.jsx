import React from 'react';
import renderer from 'react-test-renderer';
import ResultsOptions from './ResultsOptions';

test('ResultsOptions renders', () => {
  const component = renderer.create(<ResultsOptions />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
