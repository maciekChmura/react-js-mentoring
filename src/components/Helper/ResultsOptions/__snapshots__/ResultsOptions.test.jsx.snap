// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsOptions renders 1`] = `
<div
  className="ResultsOptionsStyles__ResultsOptionsWrapper-qi5qz4-0 bcyhbG"
>
  <p
    className="ResultsOptionsStyles__Display-qi5qz4-2 gCbovo"
  >
     movies found
  </p>
  <div
    className="ResultsOptionsStyles__ButtonsWrapper-qi5qz4-1 cVcvWT"
  >
    <p
      className="ResultsOptionsStyles__Display-qi5qz4-2 gCbovo"
    >
      Sort by:
    </p>
    <button
      className="OptionButtonStyles__Button-sc-1yyses4-0 hkNaKA"
      onClick={[Function]}
    >
      release date
    </button>
    <button
      className="OptionButtonStyles__Button-sc-1yyses4-0 hkNaKA"
      onClick={[Function]}
    >
      rating
    </button>
  </div>
</div>
`;
